File: lib/core/Actions.ts
export enum Action {
    SHOW_TOAST = "show_toast",
    NAVIGATE = "navigate",
    NAVIGATE_PUSH_REPLACE = "navigate_pushreplace",
    NAVIGATE_POP = "pop",
    HIDE = "hide",
    SHOW = "show",
    DEBUG = "daikon",
    DIALOG = "dialog",
    SUBMIT = "submit",
    SET_VAR = "setvar",
    JAVASCRIPT = "js",
    PROPS = "props",
    LAUNCH_URL = "launch_url",
    SELECT_FILE = "select_file"
}

export function DoAction(action: Action, data: any){
    return {
        action,
        data
    }
}

File: flutter_render/lib/src/parser/Actions.dart
import 'package:flutter/material.dart';
import 'package:neneui_render/src/enum.dart';

class ActionsPerf {
  static void perform(
    BuildContext context,
    Function event,
    String action,
    dynamic mainData, {
    Map<String, dynamic> data = const {},
  }) {
    if (action == "show_toast") {
      event(Events.INVOKE_TOAST, mainData);
    } else if (action == "navigate") {
      event(Events.INVOKE_NAVIGATE, mainData);
    } else if (action == "navigate_pushreplace") {
      event(Events.INVOKE_NAVIGATE_REPLACE, mainData);
    } else if (action == "hide") {
      event(Events.HIDE_IDB, mainData);
    } else if (action == "show") {
      event(Events.SHOW_IDB, mainData);
    } else if (action == "daikon") {
      event(Events.DAIKON_DEBUG, mainData);
    } else if (action == "setvar") {
      event(Events.SET_VAR, mainData);
    } else if (action == "dialog") {
      event(Events.DIALOG, mainData);
    } else if (action == "submit") {
      event(Events.SUBMIT, mainData);
    } else if (action == "pop") {
      event(Events.INVOKE_POP, "pop");
    } else if (action == "props") {
      event(Events.INVOKE_REPLACE_PROPS, mainData);
    } else if (action == "js") {
      event(Events.INVOKE_JS, mainData);
    } else if (action == "select_file") {
      event(Events.SELECT_FILE, mainData);
    }

    debugPrint(action);
  }
}
